web/react-web/src/Sidebar/Countries.js

```diff
@@ -0,0 +1,69 @@
+import React, {Component} from 'react';
+
+class Countries extends Component {
+
+    static defaultProps = {
+        predictions: {},
+        selections: []
+    };
+
+    toggleItem = (item) => () => {
+        const {toggle} = this.props.selections;
+        toggle(item);
+    };
+
+    render() {
+        const {
+            predictions,
+            selections
+        } = this.props;
+
+        if (!predictions.data) {
+            return null;
+        }
+
+        const countries = new Set();
+        const predictionDates = Object.keys(predictions.data).sort();
+        for (let dateKey in predictions.data) {
+            if (predictions.data.hasOwnProperty(dateKey)) {
```

Comment:
This is only version for iterating in old javascript. [source](https://zellwk.com/blog/looping-through-js-objects/)  in ES6 it is possible to to write `predictions.values.forEach(x=>{...  }) `

Reply:
> Maybe a noob js question, but why do we need to check if `dateKey` is in `data` if we are iterating over keys in `data`? And can't we iterate directly over values?

Objects could be derived. I that case, we're checking whether it's own property or derived one. This time, this could be ommited, however I was just getting warning from IDE.

> This is the only version for iterating in old javascript. [source](https://zellwk.com/blog/looping-through-js-objects/) in ES6 it is possible to to write `predictions.values.forEach(x=>{... }) `

Right, however in this case this is object. So it'd look like
```
Object.keys(yourobject).forEach((key) => {
...body...
});
```
or
```
Object.entries(predictions.data).forEach(([dateKey, prediction]) => {
  prediction.countries.forEach(country => {
    countries.add(country);
  });
});
```
I'll use later one.